hotkey_linux.go

```diff
@@ -76,6 +83,9 @@ func (hk *Hotkey) unregister() error {
 		return errors.New("hotkey is not registered.")
 	}
 	hk.cancel()
+	if hk.display != nil {
+		C.sendCancel(hk.display, hk.window)
+	}
```

Comment:
I don't feel confident about this approach. What if the display is allocated after this check? What if the Unregister happens on a different goroutine to Register? This will lead to leaking resources, and race conditions of hk.display.

I think we will need some sort of synchronization here to guarantee that unregister can be called only after display and window are allocated.